provider/source/data/debug/relativetime/long/day@1/ar-EG.json

```diff
@@ -7,55 +7,115 @@
     "2": "بعد الغد"
   },
   "past": {
-    "zero": {
-      "pattern": "قبل  يوم",
-      "index": 7
-    },
-    "one": {
-      "pattern": "قبل يوم واحد",
-      "index": 255
-    },
-    "two": {
-      "pattern": "قبل يومين",
-      "index": 255
-    },
-    "few": {
-      "pattern": "قبل  أيام",
-      "index": 7
-    },
-    "many": {
-      "pattern": "قبل  يومًا",
-      "index": 7
-    },
-    "other": {
-      "pattern": "قبل  يوم",
-      "index": 7
-    }
+    "zero": [
+      {
```

Comment:
this isn't really human-readable imho, I'd prefer if `Pattern` serialized to a pattern string instead.

Reply:
I would too, but that required adding extra trait functions to `PatternBackend` whereas serializing a list of PatternItems didn't require adding anything to the trait. I made the judgement when writing `icu_pattern` that the code cost of supporting nicer human-readable serialization was not worth the benefit.

Reply:
An easier alternative would be to clean up the serialization so long as it is still a `Vec<PatternItem>`. So for example, instead of

```json
    [
      {
        "Placeholder": "Singleton"
      },
      {
        "Literal": " yıl önce"
      }
    ]
```

you could have

```json
    [0, " yıl önce"]
```

which works if the placeholders are numeric. If the placeholders are strings, maybe something like

```json
    [{ "placeholder": "foo" }, " yıl önce"]
```

Reply:
`0` could serialize to `{0}`, and `{ "placeholder": "foo" }` could serialize to `{foo}`.

Reply:
Problem with serializing to `"{0}"` is that you can't tell if that's a placeholder or the literal string, which means we have to deal with escaping, which is a whole bunch of complexity. We should encode whether the segment is a placeholder or a string in the JSON schema.